src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  ContentChildren,
+  QueryList,
+  Renderer2,
+  ElementRef,
+  Optional,
+  SkipSelf,
+  AfterContentInit,
+  ContentChild,
+  Output,
+  AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link?: string;
+
+    @Input()
+    set level(value: number) {
+      this._level = value;
+    }
+
+    get level(): number {
+      return this._level;
+    }
+
+    @Output()
+    get active() {
+      if (this.link) {
+          return this._router.isActive(this.link, true);
+      } else {
+          return this._active;
+      }
+    }
+
+    private _level: number = 2;
+    private _active: boolean = false;
+
+    @ContentChildren(NavigationItemComponent, { descendants: true })
+    private _children: QueryList<NavigationItemComponent>;
+
+    get children(): NavigationItemComponent[] {
+      return this._children.filter(item => item !== this);
+    }    
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2,
+        @Optional()
+        @SkipSelf()
+        private _parent: NavigationItemComponent,
+        @Optional() private _router: Router
+    ) {
+        this.level = _parent ? _parent.level + 1 : 2;
+
+        if (this._router) {

```

Comment:
I don't understand this. Router is a part of Angular. Why do we need to check for it?

Reply:
Removed check for existence of router. This was a remnant of checks I was making earlier in development concerning which router objects (Router, ActivatedRoute, ...) were available in the documentation example and the Plunker example.